source/core_sntp_client.c

```diff
@@ -771,6 +762,11 @@ SntpStatus_t Sntp_ReceiveTimeResponse( SntpContext_t * pContext,
                 ( calculateElapsedTimeMs( &loopIterTime, pRequestTime ) >= pContext->responseTimeoutMs ) )
             {
                 status = SntpErrorResponseTimeout;
+
+                /* As server has timed out in sending its response, we will rotate to the next server in
+                 * the list of configured time servers. */
+                pContext->currentServerIndex++;
```

Comment:
We discussed separately, and decided to have the server rotation use wrap around once it has used up all the servers so that the SNTP client functionality is not disturbed.